File: src/ReadStorm.Domain/Models/SourceDiagnosticResult.cs
namespace ReadStorm.Domain.Models;

/// <summary>书源诊断结果，用于可视化诊断面板。</summary>
public sealed class SourceDiagnosticResult
{
    public int SourceId { get; init; }

    public string SourceName { get; set; } = string.Empty;

    public string BaseUrl { get; set; } = string.Empty;

    public bool SearchRuleFound { get; set; }

    public bool TocRuleFound { get; set; }

    public bool ChapterRuleFound { get; set; }

    public int HttpStatusCode { get; set; }

    public string HttpStatusMessage { get; set; } = string.Empty;

    public int SearchResultCount { get; set; }

    public int TocItemCount { get; set; }

    public string TocSelector { get; set; } = string.Empty;

    public string ChapterContentSelector { get; set; } = string.Empty;

    public string SampleChapterText { get; set; } = string.Empty;

    public bool IsHealthy => SearchRuleFound && TocRuleFound && ChapterRuleFound && HttpStatusCode is >= 200 and < 400;

    public string Summary { get; set; } = string.Empty;

    public List<string> DiagnosticLines { get; } = [];
}

File: pureopus/app/src/main/java/com/readstorm/app/domain/models/SourceDiagnosticResult.kt
package com.readstorm.app.domain.models

data class SourceDiagnosticResult(
    val sourceId: Int = 0,
    var sourceName: String = "",
    var baseUrl: String = "",
    var searchRuleFound: Boolean = false,
    var tocRuleFound: Boolean = false,
    var chapterRuleFound: Boolean = false,
    var httpStatusCode: Int = 0,
    var httpStatusMessage: String = "",
    var searchResultCount: Int = 0,
    var tocItemCount: Int = 0,
    var tocSelector: String = "",
    var chapterContentSelector: String = "",
    var sampleChapterText: String = "",
    var summary: String = "",
    val diagnosticLines: MutableList<String> = mutableListOf()
) {
    val isHealthy: Boolean
        get() = searchRuleFound && tocRuleFound && chapterRuleFound &&
                httpStatusCode in 200..<400
}
